Makefile

```diff
@@ -1,157 +1,69 @@
-
-	@echo "[+] Removing the Linux ARM binary"
-	@rm $(EXEC_DIR)tcping
-
-build_windows:
-	@echo
-	@echo "[+] Building the Windows version"
-	@env GOOS=windows GOARCH=amd64 go build -ldflags "-s -w" -o $(EXEC_DIR)tcping.exe
-
-	@echo "[+] Packaging the Windows version"
-	@zip -j $(EXEC_DIR)tcping_Windows.zip $(EXEC_DIR)tcping.exe > /dev/null
-	@sha256sum  $(EXEC_DIR)tcping_Windows.zip
-
-	@echo "[+] Removing the Windows binary"
-	@rm $(EXEC_DIR)tcping.exe
-
-build_macos:
-	@echo
-	@echo "[+] Building the MacOS version"
-	@env GOOS=darwin GOARCH=amd64 go build -ldflags "-s -w" -o $(EXEC_DIR)tcping
-
-	@echo "[+] Packaging the MacOS version"
-	@tar -czvf $(EXEC_DIR)tcping_MacOS.tar.gz -C $(EXEC_DIR) tcping > /dev/null
-	@sha256sum $(EXEC_DIR)tcping_MacOS.tar.gz
-
-	@echo "[+] Removing the MacOS binary"
-	@rm $(EXEC_DIR)tcping
-
-build_macos_arm:
-	@echo
-	@echo "[+] Building the MacOS ARM version"
-	@env GOOS=darwin GOARCH=arm64 go build -ldflags "-s -w" -o $(EXEC_DIR)tcping
-
-	@echo "[+] Packaging the MacOS ARM version"
-	@tar -czvf $(EXEC_DIR)tcping_MacOS_ARM.tar.gz -C $(EXEC_DIR) tcping > /dev/null
-	@sha256sum $(EXEC_DIR)tcping_MacOS_ARM.tar.gz
-
-	@echo "[+] Removing the MacOS ARM binary"
-	@rm $(EXEC_DIR)tcping
-
-build_freebsd:
-	@echo
-	@echo "[+] Building the FreeBSD version"
-	@env GOOS=freebsd GOARCH=amd64 go build -ldflags "-s -w" -o $(EXEC_DIR)tcping
-
-	@echo "[+] Packaging the FreeBSD AMD64 version"
-	@tar -czvf $(EXEC_DIR)tcping_FreeBSD.tar.gz -C $(EXEC_DIR) tcping > /dev/null
-	@sha256sum $(EXEC_DIR)tcping_FreeBSD.tar.gz
-
-	@echo "[+] Removing the FreeBSD binary"
-	@rm $(EXEC_DIR)tcping
+# ==================================================
+# Constants
+# ==================================================
+
+# Meta
+VERSION := 2.6.0
+MAINTAINER := https://github.com/pouriyajamshidi
+DESCRIPTION := Ping TCP ports using tcping. Inspired by Linux's ping utility. Written in Go
+
+# IO directories
+TARGET_DIR := target
+OUTPUT_DIR := output
+TAPE_DIR := Images/tapes
+GIFS_DIR := Images/gifs
+
+# Release outputs
+RELEASE_ARTIFACTS := \
+	$(OUTPUT_DIR)/tcping-freebsd-amd64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-freebsd-amd64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-freebsd-arm64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-freebsd-arm64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-amd64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-amd64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-arm64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-arm64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-darwin-amd64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-darwin-amd64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-darwin-arm64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-darwin-arm64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-windows-amd64-static.zip \
+	$(OUTPUT_DIR)/tcping-windows-amd64-dynamic.zip \
+	$(OUTPUT_DIR)/tcping-windows-arm64-static.zip \
+	$(OUTPUT_DIR)/tcping-windows-arm64-dynamic.zip \
+	$(OUTPUT_DIR)/tcping-amd64.deb \
+	$(OUTPUT_DIR)/tcping-arm64.deb
```

Comment:
We probably don't need all of these. Feel free to delete any that you find unnecessary.